WordPress/Classes/Stores/StatsRevampStore.swift

```diff
@@ -0,0 +1,373 @@
+    func shouldFetchViewsAndVisitors() -> Bool {
+        return viewsAndVisitorsStatus != .loading
+    }
+
+    func setViewsAndVisitorsFetchingStatus(_ status: StoreFetchingStatus) {
+        transaction { state in
+            state.summaryStatus = status
+            state.topReferrersStatus = status
+            state.topCountriesStatus = status
+        }
+    }
+
+    func fetchViewsAndVisitorsData(date: Date) {
+        loadViewsAndVisitorsCache(date: date)
+
+        guard shouldFetchViewsAndVisitors() else {
+            DDLogInfo("Stats Views and Visitors details refresh triggered while one was in progress.")
+            return
+        }
+
+        setViewsAndVisitorsFetchingStatus(.loading)
+
+        fetchSummary(date: date) { [weak self] in
+            self?.fetchViewsAndVisitorsDetailsData(date: date)
+        }
+    }
+
+    func fetchViewsAndVisitorsDetailsData(date: Date) {
+        guard let service = statsRemote() else {
+            return
+        }
+
+        let topReferrers = PeriodOperation(service: service, for: .week, date: date) { [weak self] (referrers: StatsTopReferrersTimeIntervalData?, error: Error?) in
+            if error != nil {
+                DDLogError("Stats Revamp Store: Error fetching referrers: \(String(describing: error?.localizedDescription))")
+            }
+
+            DDLogInfo("Stats Revamp Store: Finished fetching referrers.")
+
+            DispatchQueue.main.async {
+                self?.receivedReferrers(referrers, error)
+            }
+        }
+
+        let topCountries = PeriodOperation(service: service, for: .week, date: date, limit: 0) { [weak self] (countries: StatsTopCountryTimeIntervalData?, error: Error?) in
+            if error != nil {
+                DDLogError("Stats Revamp Store: Error fetching countries: \(String(describing: error?.localizedDescription))")
+            }
+
+            DDLogInfo("Stats Revamp Store: Finished fetching countries.")
+
+            DispatchQueue.main.async {
+                self?.receivedCountries(countries, error)
+            }
+        }
+
+        operationQueue.addOperations([topReferrers,
+                                      topCountries],
+                                     waitUntilFinished: false)
+    }
+
+    private func loadViewsAndVisitorsCache(date: Date) {
+        guard
+            let siteID = SiteStatsInformation.sharedInstance.siteID,
+            let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
+                return
+        }
+
+        let summary = StatsRecord.timeIntervalData(for: blog, type: .blogVisitsSummary, period: StatsRecordPeriodType(remoteStatus: .day), date: date)
+        let referrers = StatsRecord.timeIntervalData(for: blog, type: .referrers, period: StatsRecordPeriodType(remoteStatus: .week), date: date)
+        let countries = StatsRecord.timeIntervalData(for: blog, type: .countryViews, period: StatsRecordPeriodType(remoteStatus: .week), date: date)
```

Comment:
@staskus these calls here + loading data from cache are using other screens and datastore(s) such as `StatsPeriodStore` to save their data which is fine.  But I think if `StatsRevampStore` makes API calls it should also persistToCoreData as well so that its calls can be cached.

I did a test case using airplane mode where I did the following:
1. I went to Stats -> Insights -> Views and Visitors and tapped week.  I let the data load fully
2. I backed out and enabled airplane mode
3. On the Views and Visitors card I tapped week again.  This time I get a network error screen.  When I debugged this I can see the cached data for referrers and countries is nil.  I think summary data was cached because of the top level insights screen